.github/workflows/prettier.yml

```diff
@@ -0,0 +1,64 @@
+name: Format Code with Prettier
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!develop'
+      - '!master'
+      - '!single-server'
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          fetch-depth: 0
+          
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+          cache: 'yarn'
+
+      - name: Cache node_modules
+        uses: actions/cache@v4
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node-modules-
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run Prettier
+        run: yarn prettier --write .
+
+      - name: Check for changes
+        id: check_changes
+        run: |
+          if git diff --quiet; then
+            echo "No code format changes detected"
+            echo "changes=false" >> $GITHUB_OUTPUT
+          else
+            echo "Code format changes detected"
+            echo "changes=true" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Commit and push changes
+        if: steps.check_changes.outputs.changes == 'true' && github.actor != 'github-actions[bot]'
```

Comment:
You've added the condition, but the actor of the prettier commit is not `github-actions[bot]`. I just suggested that, because it works in some github actions, depending on how it's configured.
The way it is right now, it's not doing anything.

Reply:
Okay :)